provider-example/src/lib.rs

```diff
@@ -12,15 +12,10 @@ use rustls::pki_types::PrivateKeyDer;
 mod aead;
 mod hash;
 mod hmac;
-#[cfg(feature = "std")]
-mod hpke;
 mod kx;
 mod sign;
 mod verify;
 
-#[cfg(feature = "std")]
-pub use hpke::HPKE_PROVIDER;
```

Comment:
I'm also open to leaving this example in-tree if folks think that's useful. Or, we could open a PR against rustls-rustcrypto to see if they are interested in it.

The only hurdle I can think of for keeping it in-tree is finding a nice way to share the HPKE tests. I don't know that we have a great spot for integration tests that would depend on both the rustls + provider-example crates. Depending on provider-example (even as a dev-dep) in the main crate is annoying because of the large number of transitive deps (that also includes other versions of rustls).